OpenRA.Mods.Common/Traits/BotModules/McvManagerBotModule.cs

```diff
@@ -194,8 +202,13 @@ void DeployMcv(IBot bot, Actor mcv, bool move)
 					cells = cells.Shuffle(world.LocalRandom);
 
 				foreach (var cell in cells)
-					if (world.CanPlaceBuilding(cell + offset, actorInfo, bi, null))
+				{
+					// ensure there are no additional conyard near by
+					var conyardsInRange = world.FindActorsInCircle(world.Map.CenterOfCell(cell), WDist.FromCells(minRange)).Where(
```

Comment:
1. how this module behaves in other maps which has different distance between expansion?
2. if the enemy takes the expansion location, will the MCV try to expand?

Reply:
Actually, as for d2k, `McvExpansionManagerBotModule` cannot ideally handle the different distance bettween buildable islands in the sea of sand on different maps (should try all of them), so I quite confused whether `MinBaseRadius` is enough, since we all use a fixed radius in code for this issue.